influxdb3_cache/src/last_cache/mod.rs

```diff
@@ -1325,4 +1367,255 @@ mod tests {
         });
         insta::assert_json_snapshot!(caches);
     }
+
+    #[tokio::test]
+    async fn datafusion_udtf_predicate_conversion() {
```

Comment:
This test is the key change in this module, the changes above are just churn from refactoring how predicates are handled.